src/openrct2/actions/BannerPlaceAction.hpp

```diff
@@ -0,0 +1,199 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../management/Finance.h"
+#include "../world/Banner.h"
+#include "../world/MapAnimation.h"
+#include "../world/Scenery.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(BannerPlaceAction, GAME_COMMAND_PLACE_BANNER, GameActionResult)
+{
+private:
+    CoordsXYZD _loc;
+    uint8_t _bannerType{ std::numeric_limits<uint8_t>::max() };
+    uint8_t _bannerIndex{ BANNER_INDEX_NULL };
```

Comment:
It seems we have the type 'BannerIndex' for this, if we ever change the type to raise the limit would be probably better to use that.